qos/evm/http.go

```diff
@@ -15,16 +30,31 @@ var httpHeadersApplicationJSON = map[string]string{
 // an EVM-specific implementation of gateway package's HTTPResponse.
 var _ gateway.HTTPResponse = httpResponse{}
 
+// httpResponse encapsulates an HTTP response to be returned to the client
+// including payload data and status code.
 type httpResponse struct {
+	// responsePayload contains the serialized response body.
 	responsePayload []byte
+
+	// httpStatusCode is the HTTP status code to be returned.
+	// If not explicitly set, defaults to http.StatusOK (200).
+	httpStatusCode int
 }
 
+// GetPayload returns the response payload as a byte slice.
 func (hr httpResponse) GetPayload() []byte {
 	return hr.responsePayload
 }
 
+// GetHTTPStatusCode returns the HTTP status code for this response.
+// If no status code was explicitly set, returns http.StatusOK (200).
```

Comment:
This seems to deviate from the pattern of return `0` if we don't have a status code - why?

Reply:
Thanks, updated the comment to explain this. Please reopen if needed.